src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/PersonName.java

```diff
@@ -0,0 +1,128 @@
+        if (!edifactString.startsWith(PersonName.KEY_QUALIFIER)) {
+            throw new IllegalArgumentException("Can't create " + PersonName.class.getSimpleName() + " from " + edifactString);
+        }
+        return PersonName.builder()
+            .nhsNumber(getNhsNumber(edifactString))
+            .patientIdentificationType(getPatientIdentificationType(edifactString))
+            .familyName(extractNamePart("SU", edifactString))
+            .forename(extractNamePart("FO", edifactString))
+            .title(extractNamePart("TI", edifactString))
+            .middleName(extractNamePart("MI", edifactString))
+            .thirdForename(extractNamePart("FS", edifactString))
+            .build();
+    }
+
+    private static String getNhsNumber(String edifactString) {
+        String[] components = edifactString.split("\\+");
+        if (components.length > 2 && StringUtils.isNotEmpty(components[2])) {
+            return components[2].split(":")[0];
+        }
+        return null;
+    }
+
+    private static String getPatientIdentificationType(String edifactString) {
+        String[] components = edifactString.split("\\+");
+        if (StringUtils.isNotEmpty(getNhsNumber(edifactString)) && components.length > 1) {
+            return components[2].split(":")[1];
+        }
+        return null;
+    }
+
+    private static String extractNamePart(String qualifier, String text) {
+        return Arrays.stream(text.split("\\+"))
+            .filter(value -> value.startsWith(qualifier))
+            .map(value -> value.split(":")[1])
+            .findFirst()
+            .orElse(null);
+    }
+
+    @Override
+    public String getKey() {
+        return KEY;
+    }
+
+    @Override
+    public String getValue() {
+        List<String> values = new ArrayList<>();
+
+        String namesDelimiter = containsName() ? "++" : "";
+        Optional.ofNullable(this.nhsNumber)
+            .map(value -> value + ":" + this.patientIdentificationType + namesDelimiter)
+            .ifPresentOrElse(values::add, () -> values.add(namesDelimiter));
+
+        Optional.ofNullable(this.familyName)
+            .map(value -> "SU:" + value)
+            .ifPresent(values::add);
+        Optional.ofNullable(this.forename)
+            .map(value -> "FO:" + value)
+            .ifPresent(values::add);
+        Optional.ofNullable(this.title)
+            .map(value -> "TI:" + value)
+            .ifPresent(values::add);
+        Optional.ofNullable(this.middleName)
+            .map(value -> "MI:" + value)
+            .ifPresent(values::add);
+        Optional.ofNullable(this.thirdForename)
+            .map(value -> "FS:" + value)
+            .ifPresent(values::add);
+
+        return QUALIFIER + "+" + values.stream().collect(Collectors.joining("+"));
+    }
+
+    private boolean containsName() {
+        return Stream.of(this.familyName, this.forename, this.title, this.middleName, this.thirdForename)
+            .anyMatch(Objects::nonNull);
```

Comment:
What if it's empty string?

Reply:
I've added a test for this case:
    @Test
    public void When_BuildingEmptyName_Then_ReturnEmptySegment() {
        var expectedValue = "PNA+PAT+'";

        var personName = PersonName.builder()
            .build();

        assertEquals(expectedValue, personName.toEdifact());
    }

Reply:
I think they trim trailing `+`

Reply:
It's invalid with or without the trailing + so can remain as-is

Reply:
@mattd-kainos  did you mean "valid" instead?